### ISSUE TITLE
can not reproduce reusults of MSMT17

### ISSUE THREAD

**Issue by yian2271368** [KaiyangZhou/deep-person-reid#349]
hi kaiyang, 
i am trying to reproduce your work on MSMT17, the best results i got was only 73.1%(which i  used the configs you mentioned in  #257)
Can you please help me with that?

**Comment by KaiyangZhou** [comment]
pls provide
- full command
- system info
- log of arguments
... and anything else you think can help me debug

**Comment by KaiyangZhou** [comment]
for your convenience, see the log below to set the params

you'll get similar results as reported (make sure you followed the instructions on README correctly)

```bash
Show configuration
adam:
  beta1: 0.9
  beta2: 0.999
cuhk03:
  classic_split: False
  labeled_images: False
  use_metric_cuhk03: False
data:
  combineall: False
  height: 256
  norm_mean: [0.485, 0.456, 0.406]
  norm_std: [0.229, 0.224, 0.225]
  root: /vol/research/deepreid//data/reid
  save_dir: log/osnet/multiple_runs/osnet_x1_0_msmt17_flipErase_cosinelr_seed1
  sources: ['msmt17']
  split_id: 0
  targets: ['msmt17']
  transforms: ['random_flip', 'random_erase']
  type: image
  width: 128
  workers: 4
loss:
  name: softmax
  softmax:
    label_smooth: True
  triplet:
    margin: 0.3
    weight_t: 1.0
    weight_x: 0.0
market1501:
  use_500k_distractors: False
model:
  load_weights: 
  name: osnet_x1_0
  pretrained: True
  resume: 
rmsprop:
  alpha: 0.99
sampler:
  num_instances: 4
  train_sampler: RandomSampler
sgd:
  dampening: 0.0
  momentum: 0.9
  nesterov: False
test:
  batch_size: 300
  dist_metric: cosine
  eval_freq: -1
  evaluate: False
  normalize_feature: False
  ranks: [1, 5, 10, 20]
  rerank: False
  start_eval: 0
  visactmap: False
  visrank: False
  visrank_topk: 10
train:
  base_lr_mult: 0.1
  batch_size: 64
  fixbase_epoch: 10
  gamma: 0.1
  lr: 0.0015
  lr_scheduler: cosine
  max_epoch: 250
  new_layers: ['classifier']
  open_layers: ['classifier']
  optim: amsgrad
  print_freq: 20
  seed: 1
  staged_lr: False
  start_epoch: 0
  stepsize: [20]
  weight_decay: 0.0005
use_gpu: True
video:
  pooling_method: avg
  sample_method: evenly
  seq_len: 15

Collecting env info ...
** System info **
PyTorch version: 1.1.0
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX TITAN X
GPU 1: GeForce GTX TITAN X
GPU 2: GeForce GTX TITAN X
GPU 3: GeForce GTX TITAN X
GPU 4: GeForce GTX TITAN X
GPU 5: GeForce GTX TITAN X
GPU 6: GeForce GTX TITAN X

Nvidia driver version: 418.56
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] gnumpy==0.2
[pip] numpy==1.14.1
[conda] Could not collect
        Pillow (6.0.0)

Building train transforms ...
+ resize to 256x128
+ random flip
+ to torch tensor of range [0, 1]
+ normalization (mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+ random erase
Building test transforms ...
+ resize to 256x128
+ to torch tensor of range [0, 1]
+ normalization (mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
=> Loading train (source) dataset
=> Loaded MSMT17
  ----------------------------------------
  subset   | # ids | # images | # cameras
  ----------------------------------------
  train    |  1041 |    30248 |        15
  query    |  3060 |    11659 |        15
  gallery  |  3060 |    82161 |        15
  ----------------------------------------
=> Loading test (target) dataset
=> Loaded MSMT17
  ----------------------------------------
  subset   | # ids | # images | # cameras
  ----------------------------------------
  train    |  1041 |    30248 |        15
  query    |  3060 |    11659 |        15
  gallery  |  3060 |    82161 |        15
  ----------------------------------------


  **************** Summary ****************
  train            : ['msmt17']
  # train datasets : 1
  # train ids      : 1041
  # train images   : 30248
  # train cameras  : 15
  test             : ['msmt17']
  *****************************************


Building model: osnet_x1_0
Successfully loaded imagenet pretrained weights from "/user/HS128/kz00149/.cache/torch/checkpoints/osnet_x1_0_imagenet.pth"
** The following layers are discarded due to unmatched keys or layer size: ['classifier.weight', 'classifier.bias']
Model complexity: params=2,193,616 flops=978,878,352
Building softmax-engine for image-reid
=> Start training
* Only train ['classifier'] (epoch: 1/10)
Epoch: [1/250][20/472]  Time 0.116 (0.627)      Data 0.000 (0.372)      Loss 7.4978 (7.5072)    Acc 1.56 (1.25) Lr 0.001500     eta 20:32:41
Epoch: [1/250][40/472]  Time 0.120 (0.507)      Data 0.000 (0.321)      Loss 6.8805 (7.2535)    Acc 3.12 (1.72) Lr 0.001500     eta 16:36:42
Epoch: [1/250][60/472]  Time 0.116 (0.444)      Data 0.000 (0.281)      Loss 6.4697 (7.0208)    Acc 9.38 (3.33) Lr 0.001500     eta 14:32:18
Epoch: [1/250][80/472]  Time 0.115 (0.395)      Data 0.000 (0.244)      Loss 6.3236 (6.8375)    Acc 1.56 (4.38) Lr 0.001500     eta 12:56:46
Epoch: [1/250][100/472] Time 0.117 (0.361)      Data 0.000 (0.216)      Loss 6.1394 (6.6857)    Acc 14.06 (5.53)        Lr 0.001500     eta 11:48:57
Epoch: [1/250][120/472] Time 0.118 (0.336)      Data 0.000 (0.196)      Loss 6.0808 (6.5421)    Acc 4.69 (6.73) Lr 0.001500     eta 11:00:48
Epoch: [1/250][140/472] Time 0.119 (0.316)      Data 0.000 (0.179)      Loss 5.6692 (6.4247)    Acc 18.75 (7.85)        Lr 0.001500     eta 10:20:54
...
=> Final test
##### Evaluating msmt17 (source) #####
Extracting features from query set ...
Done, obtained 11659-by-512 matrix
Extracting features from gallery set ...
Done, obtained 82161-by-512 matrix
Speed: 0.1135 sec/batch
Computing distance matrix with metric=cosine ...
Computing CMC and mAP ...
** Results **
mAP: 54.9%
CMC curve
Rank-1  : 78.6%
Rank-5  : 88.1%
Rank-10 : 91.1%
Rank-20 : 93.6%
Checkpoint saved to "log/osnet/multiple_runs/osnet_x1_0_msmt17_flipErase_cosinelr_seed1/model.pth.tar-250"
Elapsed 13:25:31
```

**Comment by yian2271368** [comment]
hey, sorry for the late reply, here is the configs, i got. i believe it's the same as yours. @KaiyangZhou 
Show configuration
adam:
  beta1: 0.9
  beta2: 0.999
cuhk03:
  classic_split: False
  labeled_images: False
  use_metric_cuhk03: False
data:
  combineall: False
  height: 256
  load_train_targets: False
  norm_mean: [0.485, 0.456, 0.406]
  norm_std: [0.229, 0.224, 0.225]
  root: Z:/datasets/
  save_dir: log/osnet_x1_0_softmax_cosinelr_msmt_17V2
  sources: ['msmt17']
  split_id: 0
  targets: ['msmt17']
  transforms: ['random_flip', 'random_erase ']
  type: image
  width: 128
  workers: 4
loss:
  name: softmax
  softmax:
    label_smooth: True
  triplet:
    margin: 0.3
    weight_t: 1.0
    weight_x: 0.0
market1501:
  use_500k_distractors: False
model:
  load_weights: D:\Qiuyu\Code\OSnet_models\Osnet_V2\log\osnet_x1_0_softmax_cosinelr_msmt_17V2\model\model.pth.tar-250
  name: osnet_x1_0
  pretrained: True
  resume:
rmsprop:
  alpha: 0.99
sampler:
  num_instances: 4
  train_sampler: RandomSampler
  train_sampler_t: RandomSampler
sgd:
  dampening: 0.0
  momentum: 0.9
  nesterov: False
test:
  batch_size: 300
  dist_metric: cosine
  eval_freq: -1
  evaluate: True
  normalize_feature: False
  ranks: [1, 5, 10, 20]
  rerank: False
  start_eval: 0
  visrank: False
  visrank_topk: 10
train:
  base_lr_mult: 0.1
  batch_size: 64
  fixbase_epoch: 10
  gamma: 0.1
  lr: 0.0015
  lr_scheduler: cosine
  max_epoch: 250
  new_layers: ['classifier']
  open_layers: ['classifier']
  optim: amsgrad
  print_freq: 20
  seed: 1
  staged_lr: False
  start_epoch: 0
  stepsize: [20]
  weight_decay: 0.0005
use_gpu: True
video:
  pooling_method: avg
  sample_method: evenly
  seq_len: 15

Collecting env info ...
** System info **
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows Server 2016 Datacenter
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration:
GPU 0: Tesla V100-PCIE-32GB
GPU 1: Tesla V100-PCIE-32GB
GPU 2: Tesla V100-PCIE-32GB
GPU 3: Tesla V100-PCIE-32GB

Nvidia driver version: 441.22
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] numpydoc==0.9.1
[pip] torch==1.4.0
[pip] torch-optimizer==0.0.1a9
[pip] torchfile==0.1.0
[pip] torchnet==0.0.4
[pip] torchreid==1.2.3
[pip] torchvision==0.5.0
[conda] blas                      1.0                         mkl    defaults
[conda] mkl                       2019.4                      245    defaults
[conda] mkl-service               2.3.0            py37hb782905_0    defaults
[conda] mkl_fft                   1.0.14           py37h14836fe_0    defaults
[conda] mkl_random                1.1.0            py37h675688f_0    defaults
[conda] pytorch                   1.4.0           py3.7_cuda101_cudnn7_0    pytorch
[conda] torch-optimizer           0.0.1a9                  pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchnet                  0.0.4                    pypi_0    pypi
[conda] torchreid                 1.2.3                     dev_0    <develop>
[conda] torchvision               0.5.0                py37_cu101    pytorch
        Pillow (6.2.0)

Building train transforms ...
+ resize to 256x128
+ random flip
+ to torch tensor of range [0, 1]
+ normalization (mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
Building test transforms ...
+ resize to 256x128
+ to torch tensor of range [0, 1]
+ normalization (mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
=> Loading train (source) dataset
=> Loaded MSMT17
  ----------------------------------------
  subset   | # ids | # images | # cameras
  ----------------------------------------
  train    |  1041 |    30248 |        15
  query    |  3060 |    11659 |        15
  gallery  |  3060 |    82161 |        15
  ----------------------------------------
=> Loading test (target) dataset
=> Loaded MSMT17
  ----------------------------------------
  subset   | # ids | # images | # cameras
  ----------------------------------------
  train    |  1041 |    30248 |        15
  query    |  3060 |    11659 |        15
  gallery  |  3060 |    82161 |        15
  ----------------------------------------


  **************** Summary ****************
  source            : ['msmt17']
  # source datasets : 1
  # source ids      : 1041
  # source images   : 30248
  # source cameras  : 15
  target            : ['msmt17']
  *****************************************


Building model: osnet_x1_0
Successfully loaded imagenet pretrained weights from "C:\Users\10061570/.cache\torch\checkpoints\osnet_x1_0_imagenet.pth"
** The following layers are discarded due to unmatched keys or layer size: ['classifier.weight', 'classifier.bias']
Model complexity: params=2,193,616 flops=978,878,352
Successfully loaded pretrained weights from "D:\Qiuyu\Code\OSnet_models\Osnet_V2\log\osnet_x1_0_softmax_cosinelr_msmt_17V2\model\model.pth.tar-250"
Building softmax-engine for image-reid
##### Evaluating msmt17 (source) #####
Extracting features from query set ...
Done, obtained 11659-by-512 matrix
Extracting features from gallery set ...
Done, obtained 82161-by-512 matrix
Speed: 0.1666 sec/batch
Computing distance matrix with metric=cosine ...
Computing CMC and mAP ...
** Results **
mAP: 45.0%
CMC curve
Rank-1  : 74.0%
Rank-5  : 84.9%
Rank-10 : 88.3%
Rank-20 : 91.1%

**Comment by KaiyangZhou** [comment]
it'd be easier to read your log with ```

hmm, so if you're sure your config exactly matches the one I gave, I'd guess the reason **might** be due to the difference in data. MSMT has two versions, v1 and the later-released v2 (the difference is that faces in v2 are blurred). Which one do you use? In my case, I used the earliest version of v1 released in 2019 for my paper. Clear faces could help improve the re-id performance
